packages/web-react/src/components/Box/useBoxStyleProps.ts

```diff
@@ -0,0 +1,87 @@
+import classNames from 'classnames';
+import { CSSProperties, ElementType } from 'react';
+import { BorderColors } from '../../constants';
+import { BreakpointToken, SpaceToken, SpiritBoxProps } from '../../types';
+
+interface BoxCSSProperties extends CSSProperties {
+  [key: string]: string | undefined | number;
+}
+
+export interface UseBoxStyleProps<T> {
+  /** className props */
+  classProps: string;
+  /** Props for the box element. */
+  props: T;
+  /** Style props for the box element */
+  styleProps: BoxCSSProperties;
+}
+
+export const generateResponsiveUtilityClasses = (
+  prefix: string,
+  propValue: SpaceToken | Partial<Record<BreakpointToken, SpaceToken>> | undefined,
+): string[] => {
+  if (propValue && typeof propValue === 'object') {
+    return Object.entries(propValue).map(([breakpoint, value]) => {
+      const classValue = value?.replace('space-', '');
+
+      return breakpoint === 'mobile' ? `${prefix}-${classValue}` : `${prefix}-${breakpoint}-${classValue}`;
+    });
+  }
+
+  if (propValue && typeof propValue !== 'object') {
+    return [`${prefix}-${propValue.replace('space-', '')}`];
+  }
+
+  return [];
+};
```

Comment:
is it possible to use `generateResponsiveStylePropsClasses` function for this?

Reply:
It will be completely refactored in https://github.com/lmc-eu/spirit-design-system/pull/1831